1.0rc1: test failures in io.votable with Python 3.x and Numpy-dev

(Note, this is different from https://github.com/astropy/astropy/issues/2960 which is a different failure)

@mdboom - with 1.0rc1 and the latest Numpy dev version (as of https://github.com/numpy/numpy/commit/a02f5c8fd384b5aeffd1942747b4b6c4efdcd3a4) I am getting many similar failures in tests for ``io.votable`` with Python 3.4, such as:

```

    def test_validate():
        output = io.StringIO()
    
        # We can't test xmllint, because we can't rely on it being on the
        # user's machine.
        with catch_warnings():
            result = validate(get_pkg_data_filename('data/regression.xml'),
>                             output, xmllint=False)

astropy/io/votable/tests/vo_test.py:785: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

source = '/private/var/folders/zy/t1l3sx310d3d6p0kyxqzlrnr0000gr/T/astropy-test-tpr7jxue/lib.macosx-10.8-x86_64-3.4/astropy/io/votable/tests/data/regression.xml'
output = <_io.StringIO object at 0x109805828>, xmllint = False
filename = '/private/var/folders/zy/t1l3sx310d3d6p0kyxqzlrnr0000gr/T/astropy-test-tpr7jxue/lib.macosx-10.8-x86_64-3.4/astropy/io/votable/tests/data/regression.xml'

    def validate(source, output=None, xmllint=False, filename=None):
        """
        Prints a validation report for the given file.
    
        Parameters
        ----------
        source : str or readable file-like object
            Path to a VOTABLE_ xml file.
    
        output : writable file-like object, optional
            Where to output the report.  Defaults to ``sys.stdout``.
            If `None`, the output will be returned as a string.
    
        xmllint : bool, optional
            When `True`, also send the file to ``xmllint`` for schema and
            DTD validation.  Requires that ``xmllint`` is installed.  The
            default is `False`.  ``source`` must be a file on the local
            filesystem in order for ``xmllint`` to work.
    
        filename : str, optional
            A filename to use in the error messages.  If not provided, one
            will be automatically determined from ``source``.
    
        Returns
        -------
        is_valid : bool or str
            Returns `True` if no warnings were found.  If ``output`` is
            `None`, the return value will be a string.
        """
    
        from ...utils.console import print_code_line, color_print
    
        if output is None:
            output = sys.stdout
    
        return_as_str = False
        if output is None:
            output = io.StringIO()
    
        lines = []
        votable = None
    
        reset_vo_warnings()
    
        with data.get_readable_fileobj(source, encoding='binary') as fd:
            content = fd.read()
        content_buffer = io.BytesIO(content)
        content_buffer.seek(0)
    
        if filename is None:
            if isinstance(source, six.string_types):
                filename = source
            elif hasattr(source, 'name'):
                filename = source.name
            elif hasattr(source, 'url'):
                filename = source.url
            else:
                filename = "<unknown>"
    
        with warnings.catch_warnings(record=True) as warning_lines:
            warnings.resetwarnings()
            warnings.simplefilter("always", exceptions.VOWarning, append=True)
            try:
>               votable = parse(content_buffer, pedantic=False, filename=filename)

astropy/io/votable/table.py:261: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

source = <_io.BytesIO object at 0x109b1ca08>, columns = None, invalid = 'exception', pedantic = False, chunk_size = 256, table_number = None, table_id = None
filename = '/private/var/folders/zy/t1l3sx310d3d6p0kyxqzlrnr0000gr/T/astropy-test-tpr7jxue/lib.macosx-10.8-x86_64-3.4/astropy/io/votable/tests/data/regression.xml', unit_format = None
datatype_mapping = {}, _debug_python_based_parser = False

    def parse(source, columns=None, invalid='exception', pedantic=None,
              chunk_size=tree.DEFAULT_CHUNK_SIZE, table_number=None,
              table_id=None, filename=None, unit_format=None,
              datatype_mapping=None, _debug_python_based_parser=False):
        """
        Parses a VOTABLE_ xml file (or file-like object), and returns a
        `~astropy.io.votable.tree.VOTableFile` object.
    
        Parameters
        ----------
        source : str or readable file-like object
            Path or file object containing a VOTABLE_ xml file.
    
        columns : sequence of str, optional
            List of field names to include in the output.  The default is
            to include all fields.
    
        invalid : str, optional
            One of the following values:
    
                - 'exception': throw an exception when an invalid value is
                  encountered (default)
    
                - 'mask': mask out invalid values
    
        pedantic : bool, optional
            When `True`, raise an error when the file violates the spec,
            otherwise issue a warning.  Warnings may be controlled using
            the standard Python mechanisms.  See the `warnings`
            module in the Python standard library for more information.
            When not provided, uses the configuration setting
            ``astropy.io.votable.pedantic``, which defaults to False.
    
        chunk_size : int, optional
            The number of rows to read before converting to an array.
            Higher numbers are likely to be faster, but will consume more
            memory.
    
        table_number : int, optional
            The number of table in the file to read in.  If `None`, all
            tables will be read.  If a number, 0 refers to the first table
            in the file, and only that numbered table will be parsed and
            read in.  Should not be used with ``table_id``.
    
        table_id : str, optional
            The ID of the table in the file to read in.  Should not be
            used with ``table_number``.
    
        filename : str, optional
            A filename, URL or other identifier to use in error messages.
            If *filename* is None and *source* is a string (i.e. a path),
            then *source* will be used as a filename for error messages.
            Therefore, *filename* is only required when source is a
            file-like object.
    
        unit_format : str, astropy.units.format.Base instance or None, optional
            The unit format to use when parsing unit attributes.  If a
            string, must be the name of a unit formatter. The built-in
            formats include ``generic``, ``fits``, ``cds``, and
            ``vounit``.  A custom formatter may be provided by passing a
            `~astropy.units.UnitBase` instance.  If `None` (default),
            the unit format to use will be the one specified by the
            VOTable specification (which is ``cds`` up to version 1.2 of
            VOTable, and (probably) ``vounit`` in future versions of the
            spec).
    
        datatype_mapping : dict of str to str, optional
            A mapping of datatype names to valid VOTable datatype names.
            For example, if the file being read contains the datatype
            "unsignedInt" (an invalid datatype in VOTable), include the
            mapping ``{"unsignedInt": "long"}``.
    
        Returns
        -------
        votable : `~astropy.io.votable.tree.VOTableFile` object
    
        See also
        --------
        astropy.io.votable.exceptions : The exceptions this function may raise.
        """
        from . import conf
    
        invalid = invalid.lower()
        assert invalid in ('exception', 'mask')
    
        if pedantic is None:
            pedantic = conf.pedantic
    
        if datatype_mapping is None:
            datatype_mapping = {}
    
        config = {
            'columns'          : columns,
            'invalid'          : invalid,
            'pedantic'         : pedantic,
            'chunk_size'       : chunk_size,
            'table_number'     : table_number,
            'filename'         : filename,
            'unit_format'      : unit_format,
            'datatype_mapping' : datatype_mapping
        }
    
        if filename is None and isinstance(source, six.string_types):
            config['filename'] = source
    
        with iterparser.get_xml_iterator(
            source,
            _debug_python_based_parser=_debug_python_based_parser) as iterator:
            return tree.VOTableFile(
>               config=config, pos=(1, 1)).parse(iterator, config)

astropy/io/votable/table.py:146: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <astropy.io.votable.tree.VOTableFile object at 0x10a5487f0>, iterator = <astropy.utils.xml._iterparser.IterParser object at 0x108e32208>
config = {'_current_table_number': 1, '_warning_counts': {<class 'astropy.io.votable.exceptions.W01'>: 8, <class 'astropy.io.vo...le.exceptions.W49'>: 83, <class 'astropy.io.votable.exceptions.W10'>: 2, ...}, 'chunk_size': 256, 'columns': None, ...}

    def parse(self, iterator, config):
        config['_current_table_number'] = 0
    
        for start, tag, data, pos in iterator:
            if start:
                if tag == 'xml':
                    pass
                elif tag == 'VOTABLE':
                    if 'version' not in data:
                        warn_or_raise(W20, W20, self.version, config, pos)
                        config['version'] = self.version
                    else:
                        config['version'] = self._version = data['version']
                        if config['version'].lower().startswith('v'):
                            warn_or_raise(
                                W29, W29, config['version'], config, pos)
                            self._version = config['version'] = \
                                            config['version'][1:]
                        if config['version'] not in ('1.1', '1.2', '1.3'):
                            vo_warn(W21, config['version'], config, pos)
    
                    if 'xmlns' in data:
                        correct_ns = ('http://www.ivoa.net/xml/VOTable/v%s' %
                                      config['version'])
                        if data['xmlns'] != correct_ns:
                            vo_warn(
                                W41, (correct_ns, data['xmlns']), config, pos)
                    else:
                        vo_warn(W42, (), config, pos)
    
                    break
                else:
                    vo_raise(E19, (), config, pos)
        config['version_1_1_or_later'] = \
            util.version_compare(config['version'], '1.1') >= 0
        config['version_1_2_or_later'] = \
            util.version_compare(config['version'], '1.2') >= 0
        config['version_1_3_or_later'] = \
            util.version_compare(config['version'], '1.3') >= 0
    
        tag_mapping = {
            'PARAM'       : self._add_param,
            'RESOURCE'    : self._add_resource,
            'COOSYS'      : self._add_coosys,
            'INFO'        : self._add_info,
            'DEFINITIONS' : self._add_definitions,
            'DESCRIPTION' : self._ignore_add,
            'GROUP'       : self._add_group}
    
        for start, tag, data, pos in iterator:
            if start:
                tag_mapping.get(tag, self._add_unknown_tag)(
>                   iterator, tag, data, config, pos)

astropy/io/votable/tree.py:3342: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <astropy.io.votable.tree.VOTableFile object at 0x10a5487f0>, iterator = <astropy.utils.xml._iterparser.IterParser object at 0x108e32208>, tag = 'RESOURCE'
data = {'type': 'results'}
config = {'_current_table_number': 1, '_warning_counts': {<class 'astropy.io.votable.exceptions.W01'>: 8, <class 'astropy.io.vo...le.exceptions.W49'>: 83, <class 'astropy.io.votable.exceptions.W10'>: 2, ...}, 'chunk_size': 256, 'columns': None, ...}
pos = (16, 0)

    def _add_resource(self, iterator, tag, data, config, pos):
        resource = Resource(config=config, pos=pos, **data)
        self.resources.append(resource)
>       resource.parse(self, iterator, config)

astropy/io/votable/tree.py:3271: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <astropy.io.votable.tree.Resource object at 0x10989d7b8>, votable = <astropy.io.votable.tree.VOTableFile object at 0x10a5487f0>
iterator = <astropy.utils.xml._iterparser.IterParser object at 0x108e32208>
config = {'_current_table_number': 1, '_warning_counts': {<class 'astropy.io.votable.exceptions.W01'>: 8, <class 'astropy.io.vo...le.exceptions.W49'>: 83, <class 'astropy.io.votable.exceptions.W10'>: 2, ...}, 'chunk_size': 256, 'columns': None, ...}

    def parse(self, votable, iterator, config):
        self._votable = votable
    
        tag_mapping = {
            'TABLE'       : self._add_table,
            'INFO'        : self._add_info,
            'PARAM'       : self._add_param,
            'COOSYS'      : self._add_coosys,
            'RESOURCE'    : self._add_resource,
            'LINK'        : self._add_link,
            'DESCRIPTION' : self._ignore_add
            }
    
        for start, tag, data, pos in iterator:
            if start:
                tag_mapping.get(tag, self._add_unknown_tag)(
>                   iterator, tag, data, config, pos)

astropy/io/votable/tree.py:3114: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <astropy.io.votable.tree.Resource object at 0x10989d7b8>, iterator = <astropy.utils.xml._iterparser.IterParser object at 0x108e32208>, tag = 'TABLE', data = {'ID': 'main_table'}
config = {'_current_table_number': 1, '_warning_counts': {<class 'astropy.io.votable.exceptions.W01'>: 8, <class 'astropy.io.vo...le.exceptions.W49'>: 83, <class 'astropy.io.votable.exceptions.W10'>: 2, ...}, 'chunk_size': 256, 'columns': None, ...}
pos = (25, 0)

    def _add_table(self, iterator, tag, data, config, pos):
        table = Table(self._votable, config=config, pos=pos, **data)
        self.tables.append(table)
>       table.parse(iterator, config)

astropy/io/votable/tree.py:3071: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <[AttributeError("'Table' object has no attribute 'array'") raised in repr()] Table object at 0x1097cd358>
iterator = <astropy.utils.xml._iterparser.IterParser object at 0x108e32208>
config = {'_current_table_number': 1, '_warning_counts': {<class 'astropy.io.votable.exceptions.W01'>: 8, <class 'astropy.io.vo...le.exceptions.W49'>: 83, <class 'astropy.io.votable.exceptions.W10'>: 2, ...}, 'chunk_size': 256, 'columns': None, ...}

    def parse(self, iterator, config):
        columns = config.get('columns')
    
        # If we've requested to read in only a specific table, skip
        # all others
        table_number = config.get('table_number')
        current_table_number = config.get('_current_table_number')
        skip_table = False
        if current_table_number is not None:
            config['_current_table_number'] += 1
            if (table_number is not None and
                table_number != current_table_number):
                skip_table = True
                self._empty = True
    
        table_id = config.get('table_id')
        if table_id is not None:
            if table_id != self.ID:
                skip_table = True
                self._empty = True
    
        if self.ref is not None:
            # This table doesn't have its own datatype descriptors, it
            # just references those from another table.
    
            # This is to call the property setter to go and get the
            # referenced information
            self.ref = self.ref
    
            for start, tag, data, pos in iterator:
                if start:
                    if tag == 'DATA':
                        warn_unknown_attrs(
                            'DATA', six.iterkeys(data), config, pos)
                        break
                else:
                    if tag == 'TABLE':
                        return self
                    elif tag == 'DESCRIPTION':
                        if self.description is not None:
                            warn_or_raise(W17, W17, 'RESOURCE', config, pos)
                        self.description = data or None
        else:
            tag_mapping = {
                'FIELD'       : self._add_field,
                'PARAM'       : self._add_param,
                'GROUP'       : self._add_group,
                'LINK'        : self._add_link,
                'INFO'        : self._add_info,
                'DESCRIPTION' : self._ignore_add}
    
            for start, tag, data, pos in iterator:
                if start:
                    if tag == 'DATA':
                        warn_unknown_attrs(
                            'DATA', six.iterkeys(data), config, pos)
                        break
    
                    tag_mapping.get(tag, self._add_unknown_tag)(
                        iterator, tag, data, config, pos)
                else:
                    if tag == 'DESCRIPTION':
                        if self.description is not None:
                            warn_or_raise(W17, W17, 'RESOURCE', config, pos)
                        self.description = data or None
                    elif tag == 'TABLE':
                        # For error checking purposes
                        Field.uniqify_names(self.fields)
                        return self
    
        self.create_arrays(nrows=self._nrows, config=config)
        fields = self.fields
        names = [x.ID for x in fields]
        # Deal with a subset of the columns, if requested.
        if not columns:
            colnumbers = list(range(len(fields)))
        else:
            if isinstance(columns, six.string_types):
                columns = [columns]
            columns = np.asarray(columns)
            if issubclass(columns.dtype.type, np.integer):
                if np.any(columns < 0) or np.any(columns > len(fields)):
                    raise ValueError(
                        "Some specified column numbers out of range")
                colnumbers = columns
            elif issubclass(columns.dtype.type, np.character):
                try:
                    colnumbers = [names.index(x) for x in columns]
                except ValueError:
                    raise ValueError(
                        "Columns '%s' not found in fields list" % columns)
            else:
                raise TypeError("Invalid columns list")
    
        if not skip_table:
            for start, tag, data, pos in iterator:
                if start:
                    if tag == 'TABLEDATA':
                        warn_unknown_attrs(
                            'TABLEDATA', six.iterkeys(data), config, pos)
                        self.array = self._parse_tabledata(
>                           iterator, colnumbers, fields, config)

astropy/io/votable/tree.py:2371: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <[AttributeError("'Table' object has no attribute 'array'") raised in repr()] Table object at 0x1097cd358>
iterator = <astropy.utils.xml._iterparser.IterParser object at 0x108e32208>, colnumbers = [0, 1, 2, 3, 4, 5, ...]
fields = [<astropy.io.votable.tree.Field object at 0x1099b4470>, <astropy.io.votable.tree.Field object at 0x1099b4048>, <astrop...99a9080>, <astropy.io.votable.tree.Field object at 0x1099a9e10>, <astropy.io.votable.tree.Field object at 0x1099a9160>]
config = {'_current_table_number': 1, '_warning_counts': {<class 'astropy.io.votable.exceptions.W01'>: 8, <class 'astropy.io.vo...le.exceptions.W49'>: 83, <class 'astropy.io.votable.exceptions.W10'>: 2, ...}, 'chunk_size': 256, 'columns': None, ...}

    def _parse_tabledata(self, iterator, colnumbers, fields, config):
        # Since we don't know the number of rows up front, we'll
        # reallocate the record array to make room as we go.  This
        # prevents the need to scan through the XML twice.  The
        # allocation is by factors of 1.5.
        invalid = config.get('invalid', 'exception')
    
        # Need to have only one reference so that we can resize the
        # array
        array = self.array
        del self.array
    
        parsers = [field.converter.parse for field in fields]
        binparsers = [field.converter.binparse for field in fields]
    
        numrows = 0
        alloc_rows = len(array)
        colnumbers_bits = [i in colnumbers for i in range(len(fields))]
        row_default = [x.converter.default for x in fields]
        mask_default = [True] * len(fields)
        array_chunk = []
        mask_chunk = []
        chunk_size = config.get('chunk_size', DEFAULT_CHUNK_SIZE)
        for start, tag, data, pos in iterator:
            if tag == 'TR':
                # Now parse one row
                row = row_default[:]
                row_mask = mask_default[:]
                i = 0
                for start, tag, data, pos in iterator:
                    if start:
                        binary = (data.get('encoding', None) == 'base64')
                        warn_unknown_attrs(
                            tag, six.iterkeys(data), config, pos, ['encoding'])
                    else:
                        if tag == 'TD':
                            if i >= len(fields):
                                vo_raise(E20, len(fields), config, pos)
    
                            if colnumbers_bits[i]:
                                try:
                                    if binary:
                                        rawdata = base64.b64decode(
                                            data.encode('ascii'))
                                        buf = io.BytesIO(rawdata)
                                        buf.seek(0)
                                        try:
                                            value, mask_value = binparsers[i](
                                                buf.read)
                                        except Exception as e:
                                            vo_reraise(
                                                e, config, pos,
                                                "(in row %d, col '%s')" %
                                                (len(array_chunk),
                                                 fields[i].ID))
                                    else:
                                        try:
                                            value, mask_value = parsers[i](
                                                data, config, pos)
                                        except Exception as e:
                                            vo_reraise(
                                                e, config, pos,
                                                "(in row %d, col '%s')" %
                                                (len(array_chunk),
                                                 fields[i].ID))
                                except Exception as e:
                                    if invalid == 'exception':
                                        vo_reraise(e, config, pos)
                                else:
                                    row[i] = value
                                    row_mask[i] = mask_value
                        elif tag == 'TR':
                            break
                        else:
                            self._add_unknown_tag(
                                iterator, tag, data, config, pos)
                        i += 1
    
                if i < len(fields):
                    vo_raise(E21, (i, len(fields)), config, pos)
    
                array_chunk.append(tuple(row))
                mask_chunk.append(tuple(row_mask))
    
                if len(array_chunk) == chunk_size:
                    while numrows + chunk_size > alloc_rows:
                        alloc_rows = self._resize_strategy(alloc_rows)
                    if alloc_rows != len(array):
                        array = _resize(array, alloc_rows)
                    array[numrows:numrows + chunk_size] = array_chunk
                    array.mask[numrows:numrows + chunk_size] = mask_chunk
                    numrows += chunk_size
                    array_chunk = []
                    mask_chunk = []
    
            elif not start and tag == 'TABLEDATA':
                break
    
        # Now, resize the array to the exact number of rows we need and
        # put the last chunk values in there.
        alloc_rows = numrows + len(array_chunk)
    
>       array = _resize(array, alloc_rows)

astropy/io/votable/tree.py:2521: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

masked = masked_recarray(data = [],
                mask = [],
          fill_value = ('?', b'N/A', '?', 'N/A', b'N/A', 63, 169...ray', '<i4', (2, 2)), ('precision1', '<f8'), ('precision2', '<f8'), ('doublearray', 'O'), ('bitarray2', '?', (16,))]))

new_size = 5

    def _resize(masked, new_size):
        """
        Masked arrays can not be resized inplace, and `np.resize` and
        `ma.resize` are both incompatible with structured arrays.
        Therefore, we do all this.
        """
        new_array = ma.zeros((new_size,), dtype=masked.dtype)
        length = min(len(masked), new_size)
>       new_array.data[:length] = masked.data[:length]

astropy/io/votable/tree.py:70: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = rec.array([], 
          dtype=[(('string test', 'string_test'), '|O8'), (('fixed string test', 'string_test_2'), '|S1...y', '<i4', (2, 2)), ('precision1', '<f8'), ('precision2', '<f8'), ('doublearray', '|O8'), ('bitarray2', '|b1', (16,))])
indx = slice(None, 0, None)

    def __getitem__(self, indx):
        obj = ndarray.__getitem__(self, indx)
    
        # copy behavior of getattr, except that here
        # we might also be returning a single element
        if isinstance(obj, ndarray):
            if obj.dtype.fields:
>               return obj.view(dtype=(self.dtype.type, obj.dtype.descr))

/Users/tom/Library/Python/3.4/lib/python/site-packages/numpy/core/records.py:481: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = rec.array([], 
          dtype=[(('string test', 'string_test'), '|O8'), (('fixed string test', 'string_test_2'), '|S1...y', '<i4', (2, 2)), ('precision1', '<f8'), ('precision2', '<f8'), ('doublearray', '|O8'), ('bitarray2', '|b1', (16,))])
dtype = dtype((numpy.record, [(('string test', 'string_test'), 'O'), (('fixed string test', 'string_test_2'), 'S10'), ('unicod...rray', '<i4', (2, 2)), ('precision1', '<f8'), ('precision2', '<f8'), ('doublearray', 'O'), ('bitarray2', '?', (16,))]))
type = None

    def view(self, dtype=None, type=None):
        if dtype is None:
            return ndarray.view(self, type)
        elif type is None:
            try:
                if issubclass(dtype, ndarray):
                    return ndarray.view(self, dtype)
            except TypeError:
                pass
            dtype = sb.dtype(dtype)
            if dtype.fields is None:
                return self.__array__().view(dtype)
>           return ndarray.view(self, dtype)

/Users/tom/Library/Python/3.4/lib/python/site-packages/numpy/core/records.py:540: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = rec.array([], 
          dtype=[(('string test', 'string_test'), '|O8'), (('fixed string test', 'string_test_2'), '|S1...y', '<i4', (2, 2)), ('precision1', '<f8'), ('precision2', '<f8'), ('doublearray', '|O8'), ('bitarray2', '|b1', (16,))])
attr = 'dtype'
val = dtype((numpy.record, [(('string test', 'string_test'), 'O'), (('fixed string test', 'string_test_2'), 'S10'), ('unicod...rray', '<i4', (2, 2)), ('precision1', '<f8'), ('precision2', '<f8'), ('doublearray', 'O'), ('bitarray2', '?', (16,))]))

    def __setattr__(self, attr, val):
        newattr = attr not in self.__dict__
        try:
            ret = object.__setattr__(self, attr, val)
        except:
            fielddict = ndarray.__getattribute__(self, 'dtype').fields or {}
            if attr not in fielddict:
                exctype, value = sys.exc_info()[:2]
>               raise exctype(value)
E               TypeError: Cannot change data-type for object array.

/Users/tom/Library/Python/3.4/lib/python/site-packages/numpy/core/records.py:457: TypeError
```

--- a/astropy/io/votable/tree.py
+++ b/astropy/io/votable/tree.py
@@ -67,9 +67,15 @@ def _resize(masked, new_size):
     """
     new_array = ma.zeros((new_size,), dtype=masked.dtype)
     length = min(len(masked), new_size)
-    new_array.data[:length] = masked.data[:length]
-    if length != 0:
-        new_array.mask[:length] = masked.mask[:length]
+    try:
+        # Pre Numpy 1.10 way
+        new_array.data[:length] = masked.data[:length]
+    except TypeError:
+        # Numpy 1.10 and later
+        new_array[:length] = masked[:length]
+    else:
+        if length != 0:
+            new_array.mask[:length] = masked.mask[:length]
     return new_array
 
 

